test/loader.unittest.ts

```diff
@@ -0,0 +1,161 @@
+
+    // Config specifies that window shouldn't be registered
+    const config = getWindowConfig(store, app.identity);
+
+    await assert.strictEqual(config.enabled, false);
+});
+
+it('Config is unloaded when the application exits', async () => {
+    const app = createAppWithConfig(createUuid(), {enabled: false});
+
+    // Let the loader catch up
+    await delay(10);
+
+    // Sanity check - make sure config was definitely loaded initially
+    const preConfig = getWindowConfig(store, app.identity);
+    assert.strictEqual(preConfig.enabled, false);
+
+    await app.close();
+
+    // App-specific config has been removed, querying 'enabled' returns the default value of true
+    const postConfig = await getWindowConfig(store, app.identity);
+    assert.strictEqual(postConfig.enabled, true);
+});
+
+it('If an application creates a child application, the config of the parent application persists for the lifecycle of its child', async () => {
+    const app = createAppWithConfig(createUuid(), {enabled: false});
+    const child = createAppWithConfig(createUuid(), {}, app.identity.uuid);
+
+    // Let the loader catch up
+    await delay(10);
+
+    // Config should disable main app, child app remains registered
+    assert.strictEqual((await getWindowConfig(store, app.identity)).enabled, false);
+    assert.strictEqual((await getWindowConfig(store, child.identity)).enabled, true);
+
+    await app.close();
+
+    // No change in config state, as child app extends the lifespan of main app's config
+    assert.strictEqual((await getWindowConfig(store, app.identity)).enabled, false);
+    assert.strictEqual((await getWindowConfig(store, child.identity)).enabled, true);
+
+    await child.close();
+
+    // Config should now revert to initial state (everything enabled)
+    assert.strictEqual((await getWindowConfig(store, app.identity)).enabled, true);
+    assert.strictEqual((await getWindowConfig(store, child.identity)).enabled, true);
+});
+
+
+it('If an application creates a child application, the parent can apply rules to the child that still apply after the parent exits', async () => {
+    const childAppUuid = createUuid();
+
+    const app = await createAppWithConfig(createUuid(), {enabled: false,
+        rules: [{scope: {level: 'application', uuid: childAppUuid}, config: {features: {featureOne: false}}}]
+    });
+    const childApp = createAppWithConfig(childAppUuid, {}, app.identity.uuid);
+
+    const childWindow = childApp.createChildWindow('child-window-1');
+
+    // Let the loader catch up
+    await delay(10);
+
+    // Close parent app, small delay to ensure loader captures events
+    await app.close();
+    await delay(10);
+
+    assert.strictEqual(getWindowConfig(store, childApp.identity).features.featureOne, false);
+    assert.strictEqual(getWindowConfig(store, childWindow.identity).features.featureOne, false);
+});
+
+
+it('For two independent applications, there is no shared lifecycle', async () => {
```

Comment:
It was removed because its essentially the same test as `For two independent applications, there is no shared lifecycle` behind the scenes.  When you call `createFromManifest()` from anywhere, there is no connection between the newly created app and the app that called it, as in they both become two independent applications. To mock it would be just be creating a new `FakeApplication` which seems a bit silly to do when we have a test that does just that.

Reply:
👍